test/e2e/provisioning/all_e2e_test.go

```diff
@@ -388,6 +389,30 @@ func TestAWSProvisioningE2E(t *testing.T) {
 	runScenarios(t, selector, params, AWSManifest, fmt.Sprintf("aws-%s", *testRunIdentifier))
 }
 
+// TestAWSAssumeRoleProvisioning - a test suite that exercises AWS provider
+// by requesting nodes using an assumed role.
+func TestAWSAssumeRoleProvisioningE2E(t *testing.T) {
+	t.Parallel()
+
+	// test data
+	awsKeyID := os.Getenv("AWS_E2E_TESTS_KEY_ID")
+	awsSecret := os.Getenv("AWS_E2E_TESTS_SECRET")
+	awsAssumeRoleARN := os.Getenv("AWS_E2E_ASSUME_ROLE_ARN")
+	awsAssumeRoleExternalID := os.Getenv("AWS_E2E_ASSUME_ROLE_EXTERNAL_ID")
+	if len(awsKeyID) == 0 || len(awsSecret) == 0 {
+		t.Fatal("unable to run the test suite, AWS_E2E_TESTS_KEY_ID or AWS_E2E_TESTS_SECRET environment variables cannot be empty")
+	}
+	selector := Not(OsSelector("sles"))
```

Comment:
You probably do not need to select everything, ubuntu can be enough. In other words instead of calling`runScenarios` call `testScenario` similar to flatcar setup. https://github.com/kubermatic/machine-controller/blob/master/test/e2e/provisioning/all_e2e_test.go#L490-L497